lib/cmd/common/build.js

```diff
@@ -48,6 +48,7 @@ function build(argv, cb) {
     validateArgs(argObj);
     if (isNGUI()) {
       validateArgsNgui(argObj);
+      checkTag(argObj,cb);
```

Comment:
the `checkTag` is an async function call. If it's invoked like this, the functions from L53-54 will also be invoked as well, even if `checkTag` is failed. 

The right way to do it is to call the next 2 functions only if `checkTag` is passed,  something like this:

```javascript
checkTag(argObj, function(err){
  if (err) {
    return cb(err);
  } else {
     mutateGitRefArgs(argObj);
     doBuild(argObj, cb);
  }
});
```